src/main/perf/IndexAndSearchOpenStreetMaps.java

```diff
@@ -935,11 +935,6 @@ public void collect(int doc) {
                     Sort sort = new Sort(LatLonDocValuesField.newDistanceSort("point", centerLat, centerLon));
                     for(IndexSearcher s : searchers) {
                       TopFieldDocs hits = s.search(q, 10, sort);
-                      if (hits.totalHits.relation != TotalHits.Relation.EQUAL_TO) {
-                    	  // IndexSearcher can never optimize top-hits collection in that case,
-                    	  // se we should get accurate hit counts
-                    	  throw new AssertionError();
-                      }
                       totHits += hits.totalHits.value;
```

Comment:
We should do something about `totHits` too?

Reply:
yeah, total hits just tell us how many docs we have visited and we use it to compute the number of documents per second. Not sure exactly what we can do.

Reply:
maybe we can do like we do for nearest neigbors: in the final message we replace totHits with the number that really matters, e.g. in this case `searcher.count(q)`?

Reply:
I'm glad to see we are tripping this `AssertionError`!  It means Lucene was indeed able to optimize this case thanks to LUCENE-9449, yay!

Yeah I don't think just removing the `AssertionError` check is enough, because the `hits.totalHits` will typically be a (often big) undercount of how many possible hits were "considered".  Though, it is a fair count of how many hits it did in fact check?

Using `searcher.count(q)` seems good, except, we need this call to not be included in the benchmark runtime.  I think what we need is a loop outside of the `tStart - tEnd` runtime, that sums up the `searcher.count(q)` for all queries tested.  Or, maybe, we need to somehow disable this opto, so we can again test the raw performance of the query, forcing it to visit all matches?  Seems not natural though.

Maybe we can add one `ITER`, and use that last iteration to do the counting, while skipping the timing measurement for it.

Reply:
I added a list to collect the queries in case of `doDistanceSort`. Then before computing the result, we do the actual count using the collected queries. wdyt?